### DISCUSSION TITLE
[dropdown] Searching dropdown full-text returns weird matches

### DISCUSSION THREAD

**Discussion by fgeorges** [fomantic/Fomantic-UI#1737] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
Hi,

I am confused by the full-text search on dropdowns.  For instance, searching for `hope` in the [first example](https://jsfiddle.net/0hsrpzdu/) below will return `A New Hope` (OK), `T-16 skyhopper` (close enough), and `Zephyr-G swoop bike`.  That last one is a complete mystery to me.

<img width="464" alt="Screen Shot 2020-11-01 at 15 58 00" src="https://user-images.githubusercontent.com/584411/97806272-24cdc980-1c5b-11eb-8fed-d3a0a08b2c93.png">

If I [replace the values](https://jsfiddle.net/f5zw8s9y/) with a value property different than its name, I cannot make any sense anymore of the same search.

Any idea what's the logic behind these results?

**Code**
- https://jsfiddle.net/0hsrpzdu/: values like `{value: "A New Hope", name: "A New Hope"}`
- https://jsfiddle.net/f5zw8s9y/: values like `{value: "http://h2o.consulting/ns/star-wars#film-1", name: "A New Hope"}`
- https://gist.github.com/fgeorges/fd48a4809408c1b02ac7192a3c79aec8: complete code

**Comment by lubber-de** [comment, marked as answer]
Use `match: 'text'` to point to the matching field
See https://jsfiddle.net/lubber/2wu3eg67/1/

**Reply by fgeorges** [reply]
Thank you for your quick answer!

Unfortunately, as far as I understand, it only makes my second example behaves like my first example (because it restricts search to match on one field instead of two, and the example 1 has the same value for both fields.)

But the behaviour puzzles my for the first example as well.  How is "`Zephyr-G swoop bike`" matching "`hope`"?

And even for the example 2, the values all being like "`http://h2o.consulting/ns/star-wars#film-1`", I don't see either how they match "`hope`" as well.

Any idea?  Is the matching algorithm documented somewhere?

**Reply by ko2in** [reply]
According to this [line](https://github.com/fomantic/Fomantic-UI/blob/develop/src/definitions/modules/dropdown.js#L901), if you have `fullTextSearch: true` it'll use `fuzzySearch` method which checks whether the search term has the same characters as the query word. So, `Zephyr-G swoop bike` contains `h`, `o`, `p` and `e`, and returns the term `Zephyr-G swoop bike` in the result list.

The same for your example 2: there are `h` and `o` in `http://h2o.consulting/ns/star-wars#film-1`.

I think `fullTextSearch: 'exact'` would be the solution for your use case.

**Reply by fgeorges** [reply]
That's it!  Now it makes sense (in all 3 cases, `false`, `true` and `'exact'`.)  Thank you for the answer!

It is clearly stated in the documentation, and I remember I did read it at some point.  So the problem was somewhere between my chair and my keyboard, where the info got lost somehow.  Sorry for the noise...

**Reply by fgeorges** [reply]
For the record, the answer is in @ko2in 's comment (but a comment cannot be marked itself as the answer...)
